Scripts/Editor/ThrusterEditor.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2022 Laboratory for Underwater Systems and Technologies (LABUST)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#if UNITY_EDITOR
+
+using UnityEditor;
+using UnityEngine;
+using Marus.Actuators;
+using System.Linq;
+using System.Collections.Generic;
+using System;
+
+namespace Marus.Sensors
+{
+    /// <summary>
+    /// Custom editor for Thruster component.
+    /// Enables thruster configuration loading, saving and modifying.
+    /// </summary>
+    [CustomEditor(typeof(Thruster))]
+    public class ThrusterEditor : Editor
+    {
+        SerializedObject ThrusterSO;
+        Thruster myThruster;
+        bool disableSaving = true;
+        List<string> thursterNames;
+        string newThrusterName;
+        string thrusterFolderPath = "Assets/marus-core/Datasheets/";
+        List<ThrusterAsset> thrusters;
+        int previousThrusterIndex = -1;
+        int selectedThrusterIndex = 0;
+        AnimationCurve currentCurve = new AnimationCurve();
+
+        void OnEnable()
+        {
+            ThrusterSO = new SerializedObject(target);
+            myThruster = (Thruster)target;
+            thrusters = GetAllInstances<ThrusterAsset>().ToList();
```

Comment:
What happens when you add new ThrusterAsset? Does this class register it and calls OnEnable() again?

Reply:
Yes. I tested it, it works well.